sample_consensus/include/pcl/sample_consensus/impl/sac_model_plane.hpp

```diff
@@ -245,7 +245,88 @@ pcl::SampleConsensusModelPlane<PointT>::countWithinDistance (
     PCL_ERROR ("[pcl::SampleConsensusModelPlane::countWithinDistance] Invalid number of model coefficients given (%lu)!\n", model_coefficients.size ());
     return (0);
   }
+#if defined (__AVX__) && defined (__AVX2__)
+  return countWithinDistanceAVX (model_coefficients, threshold);
+#elif defined (__SSE__) && defined (__SSE2__) && defined (__SSE4_1__)
+  return countWithinDistanceSSE (model_coefficients, threshold);
+#else
+  return countWithinDistanceStandard (model_coefficients, threshold);
+#endif
+}
+
+//////////////////////////////////////////////////////////////////////////
+template <typename PointT> std::size_t
+pcl::SampleConsensusModelPlane<PointT>::countWithinDistanceStandard (
+      const Eigen::VectorXf &model_coefficients, const double threshold) const
+{
+  std::size_t nr_p = 0;
+  std::size_t i = 0;
+  // Iterate through the 3d points and calculate the distances from them to the plane
+  for (; i < indices_->size (); ++i)
```

Comment:
I wouldn't move them out of the class since they logically belong there. They also need access to the `input_`-member. Having an indices-parameter would IMO complicate things unnecessarily, and would also not really help with the "rest"-points in the SIMD implementations, since then they would have to construct a vector with the remaining indices (up to 3 or 7, respectively).

Reply:
Yeah, that would be unweildy. Forgot that indices don't refer to the points themselves (like iterator or `std::reference_wrapper`)